src/components/cart/cart-view/cart-view.tsx

```diff
@@ -14,6 +14,8 @@ export interface CartViewProps {
     onViewCart: () => void;
     onItemQuantityChange: (args: { id: string; quantity: number }) => void;
     onItemRemove: (id: string) => void;
+    isCartItemUpdating?: (id: string) => boolean;
```

Comment:
there are two props (`isCartItemUpdating` and `isCartUpdating`) named almost exactly the same but one of the is a function and another - boolean. Let's maybe rename this to `checkIfCartItemUpdating` or something in this manner.

another think to consider is just passing `updatingItemIds`.. I don't have strong arguments for that but just think that props that are primitives (or array of primitives) are easier to grasp than functions.